analysis/paladinholy/src/modules/shadowlands/legendaries/MaraadsCastRatio.tsx

```diff
@@ -20,23 +32,48 @@ class MaraadsCastRatio extends Analyzer {
       this.LOTMcast,
     );
     this.addEventListener(
-      Events.cast.spell(SPELLS.LIGHT_OF_DAWN_CAST).by(SELECTED_PLAYER),
-      this.LODcast,
+      Events.applybuff.spell(SPELLS.MARAADS_DYING_BREATH_BUFF).by(SELECTED_PLAYER),
+      this.MaraadsBuffAdded,
+    );
+    this.addEventListener(
+      // this fires every time, whether its removed by LOTM, naturally expired, or overwritten by another LOD cast (for some reason)
+      Events.removebuff.spell(SPELLS.MARAADS_DYING_BREATH_BUFF).by(SELECTED_PLAYER),
+      this.MaraadsBuffRemoved,
+    );
+    this.addEventListener(
+      // this only fires when the Maraads buff is removed by LOTM, not when it naturally expires or is overwritten.
+      Events.removebuffstack.spell(SPELLS.MARAADS_DYING_BREATH_BUFF).by(SELECTED_PLAYER),
+      this.MaraadsBuffStacksRemoved,
     );
   }
 
   LOTMcast(event: CastEvent) {
-    this.LOTMCasts += 1;
+    this.totalCasts += 1;
+    if (this.maraadsBuffed === true) {
+      this.maraadsBuffed = false;
+    } else {
+      this.unbuffedCasts += 1;
+    }
```

Comment:
This can be simplified to:

```suggestion
    if (!this.maraadsBuffed) {
        this.unbuffedCasts += 1;
    }
    this.maraadsBuffed = false;
```

since maraads buff will always be consumed by LotM. If the buff isn't present, nothing changes. If the buff is present, it gets reset.